Allow comment in the same line as values

fiolj opened the pull request:
In its present form the library allows comments in the same line that are defined the pairs 'key=values'
but it does not work for interpolation. The very small change proposed would allow that.

The proposed change will fix the behavior, for the fail in the following (modified) example:

```
[DEFAULTS]
path = ../include        # Default Path to files
use_abs = True

[Section 1]
nmax = 30
# comment 1
vmin = 1.0
freqs = 5.0, 10.0, 30.0, 50.0
amps = 0.0, 1.0, 1.0, 0.0
path = ../text          # Path to files

[Section 2]
use_abs = no
; comment 2
my file = ${Section 1:path}/file.txt
```

pkgpl reviewed (approved):
Thank you.

pkgpl reviewed (changes requested):
Sorry, this modification could not pass test - test/test_intpl.e

fiolj commented:
Hi, thanks for trying the change.
I should have tested it more thoroughly.
The changes will require a few more line of code.

If you are interested in implementing comments on the same line, I've now added a function to remove comments.
The problem appears if a comment char (i.e: "#" or ";") appears as part of a string value.

Additionally I've found that tabs are not removed (because trim only works with spaces).
I've added a function "strip" to take care of that.

Please tell me if you are interested on trying those changes, and I will correct the pull request.

Thanks for all the work!

pkgpl commented:
Thank you very much for your contribution. I will merge your request.
